doc/posts/2023/2023_06_19_vara_week4.rst

```diff
@@ -0,0 +1,18 @@
+Diffusion research continues: Week 4
+====================================
+
+.. post:: June 19, 2023
+     :author: Vara Lakshmi Bayanagari
+     :tags: google
+     :category: gsoc
+
+
+What I did this week
+~~~~~~~~~~~~~~~~~~~~
+As discussed last week, I completed researching on StableDiffusion. Currently we're looking for unconditional image reconstrcution/denoising/generation using SD. I completed putting together keras implementation of unconditional SD. Since I couldn't find official implementation of unconditional SD code, I collated DDPM codebase, VQ-VAE codebase seperately. DDPM code uses Attention based U-Net for noise prediction.The basic code blocks of this U-Net are ResidualBlock & AttentionBlock. Downsampling & Upsampling are performed 4 times with decreasing & increasing width respectively. At each downsampling layer, there's two Residual Blocks followed by an optional Attention Block, keras.downsample layer on input image and timestep. At each upsampling layer, there's a concatenation from respective downsampling layer, three Residual Blocks and an optional Attention Block, keras.upsample layer. At the Middle layer, there's two ResidualBlocks with an AttentionBlock in between resulting in no change in the output size. The final ouput of Upsampling layer is followed by an GroupNormalization layer, Swich Activation layer and Conv2D layer to provide an output with desired dimesions.
```

Comment:
Some say Residual Block and some say ResdualBlock. Would be nice if you could check the overall terminologies are consistent. Also, instead of just saying keras.downsample layer, it would be better if you are more precise. Was it max pooling? Convolution with strides? Same goes for upsample layer and residual/attention blocks.